# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/backgroundhangmonitor/ThreadStackHelper.h
# Commit: bf92aff87824
# Full Hash: bf92aff878241e4f9e2b6df2a74f636685f7a3d5
# Author: Nazım Can Altınova <canaltinova@gmail.com>
# Date: 2025-05-31 08:59:27
# Description:
#   Bug 1966705 - Disable BHR on 32 bit Android due to EHABI stackwalker crashes r=mstange
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251923
# ==============================================================================

diff -r 7d5de260f508 -r bf92aff87824 toolkit/components/backgroundhangmonitor/ThreadStackHelper.h
--- a/toolkit/components/backgroundhangmonitor/ThreadStackHelper.h	Fri May 30 18:39:02 2025 +0000
+++ b/toolkit/components/backgroundhangmonitor/ThreadStackHelper.h	Fri May 30 18:42:56 2025 +0000
@@ -33,6 +33,13 @@
 #    define MOZ_THREADSTACKHELPER_NATIVE_STACK
 #  endif
 
+// There are frequent crashes on Android 32 bit ARM builds during EHABI
+// stackwalking. See bug 1969481.
+#  if defined(__ANDROID__) && defined(__arm__)
+#    undef MOZ_THREADSTACKHELPER_PROFILING_STACK
+#    undef MOZ_THREADSTACKHELPER_NATIVE_STACK
+#  endif
+
 namespace mozilla {
 
 /**